src/overlays/actors/ovl_En_Po_Fusen/z_en_po_fusen.c

```diff
@@ -21,34 +29,230 @@ const ActorInit En_Po_Fusen_InitVars = {
+    this->actor.world.pos.y += heightOffset;
+    this->actor.shape.rot.z =  (Math_Sins(this->randBaseRotChange) * 910.0f);
+    if ((this->randScaleChange < 0x4000) && (this->randScaleChange >= -0x3FFF)) {
+        Math_SmoothScaleMaxMinS( &this->limb9Rot, 0x38E, 0x14, 0xBB8, 0x64);
+    } else {
+        Math_SmoothScaleMaxMinS( &this->limb9Rot, 0x71C, 0x8, 0xBB8, 0x64);
+    }
+    this->avgBaseRotation = this->limb9Rot * 3;
+    this->limb3Rot  = (Math_Sins(this->randBaseRotChange + 0x38E3) * this->avgBaseRotation);
+    this->limb46Rot = (Math_Sins(this->randBaseRotChange)          * this->avgBaseRotation);
+    this->limb57Rot = (Math_Sins(this->randBaseRotChange - 0x38E3) * this->avgBaseRotation);
+    this->limb8Rot  = (Math_Sins(this->randBaseRotChange - 0x71C6) * this->avgBaseRotation);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/EnPoFusen_Destroy.asm")
+    shadowScaleTmp = ((1.0f - Math_Sins(this->randScaleChange)) * 10.0f) + 50.0f;
+    shadowAlphaTmp = ((1.0f - Math_Sins(this->randScaleChange)) * 75.0f) + 100.0f;
+    this->actor.shape.shadowScale = shadowScaleTmp;
+    this->actor.shape.shadowAlpha = (shadowAlphaTmp > f255) ? (u8) f255 : (u8) shadowAlphaTmp;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC1450.asm")
+void EnPoFusen_IncrementMalonPop(EnPoFusen *this) {
+    Actor* parent = this->actor.parent;
+    EnMa4* romani;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC14A4.asm")
+    if ((parent != 0) && (parent->id == ACTOR_EN_MA4)) { 
+        romani = (EnMa4*) parent;
+        romani->unk338 += 1;
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC1574.asm")
+    this->actor.draw = NULL;
+    this->actionFunc = EnPoFusen_Pop;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC158C.asm")
+void EnPoFusen_Pop(EnPoFusen *this, GlobalContext *gCtx) {
+    Actor_Spawn(&gCtx->actorCtx, gCtx, ACTOR_EN_CLEAR_TAG,
+       this->actor.world.pos.x, this->actor.world.pos.y + 20.0f, this->actor.world.pos.z, 
+       255, 255, 200, 2);
+    func_800B8EC8(&this->actor, 0x180E); // NA_SE_IT_BOMB_EXPLOSION sfx
+    Actor_MarkForDeath(&this->actor);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC18EC.asm")
+void EnPoFusen_InitFuse(EnPoFusen *this) {
+    s16 rotZ = this->actor.shape.rot.z;
+    this->fuse = GET_FUSE_LEN_PARAM(this);
+    this->actor.shape.rot.z = 0;
+    this->randScaleChange = rotZ & 0xFFFFu;
+    this->actionFunc = EnPoFusen_IdleFuse;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC192C.asm")
+void EnPoFusen_IdleFuse(EnPoFusen *this, GlobalContext *gCtx) {
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC19B0.asm")
+    EnPoFusen_Idle(this, gCtx);
+    if ((this->fuse--) == 0) {
```

Comment:
pre-decrement --this->fuse, gets around the bracket.
Could do ! instead of == 0 too

Reply:
![postdecrement](https://user-images.githubusercontent.com/42048411/112209261-807efc00-8bd6-11eb-879f-1866de43f838.png)